WordPress/openverse
# Expand documentation on API media properties

## Fixes



Fixes #4024 by @dhruvkb

## Description



This PR fills out documentation information for nearly all the API fields. Where possible, I've added more information about what should be contained in certain `JSONField`s.

I also added some automatic linking of related model docstrings to the generated documentation output. It seemed prudent to leverage that over repeating the documentation as help text which may go out of date.

A migration has been added for this change, but per `sqlmigrate api 0060`, it's all no-ops:

```
BEGIN;
--
-- Alter field alt_files on audio
--
-- (no-op)
--
-- Alter field last_synced_with_source on audio
--
-- (no-op)
--
-- Alter field meta_data on audio
--
-- (no-op)
--
-- Alter field removed_from_source on audio
--
-- (no-op)
--
-- Alter field tags on audio
--
-- (no-op)
--
-- Alter field view_count on audio
--
-- (no-op)
--
-- Alter field watermarked on audio
--
-- (no-op)
--
-- Alter field last_synced_with_source on image
--
-- (no-op)
--
-- Alter field meta_data on image
--
-- (no-op)
--
-- Alter field removed_from_source on image
--
-- (no-op)
--
-- Alter field tags on image
--
-- (no-op)
--
-- Alter field view_count on image
--
-- (no-op)
--
-- Alter field watermarked on image
--
-- (no-op)
COMMIT;
```

## Testing Instructions


Check out the rendered document and make sure all the new additions make sense and are correct!


## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
